catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/SearchFormsApplication.java

```diff
@@ -113,6 +118,18 @@ public void init() {
         APPLICATION_JSON,
         (req, res) -> {
           String id = req.params(":id");
+          Subject subject = ThreadContext.getSubject();
+          String currentUser = SubjectUtils.getName(subject);
+
+          Map<String, Object> originalMetacardOwner =
+              JsonFactory.create().parser().parseMap(util.safeGetBody(req));
+
+          if (!originalMetacardOwner.get(Core.METACARD_OWNER).equals(currentUser)) {
```

Comment:
✏️ It would be nice to have more complex decision logic here involving user roles and such, but that can be left for future improvements

Reply:
I agree. I'm going to create a ticket in the next sprint to improve all the filtering logic for roles and individuals to be improved on the backend, which will also remove the need to do any role filtering on the UI.